symengine/pow.cpp

```diff
@@ -147,6 +147,9 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
             ->power_num(outArg(coef), d, rcp_static_cast<const Number>(b));
         return Mul::from_dict(coef, std::move(d));
     }
+    if (is_a<const Log>(*b) and eq(*a, *E))
+        return rcp_static_cast<const Log>(b)->get_arg();
```

Comment:
To enforce this, there are some other places where you have to check this. For example, `E**(log(x) + 1) / E` should not be `E**log(x)`, but `x`